tools/deploy-api.ps1

```diff
@@ -0,0 +1,255 @@
+	.PARAMETER app_version
+	The version of the API for the application. Required.
+	.PARAMETER site_hostname
+	The hostname of te API website. Required.
+	.PARAMETER code_folder
+	The folder from which to copy the latest code. Required.
+#>
+
+Param(
+	[Parameter(mandatory=$true)]
+	[string]$site_name = "",
+	
+	[Parameter(mandatory=$true)]
+	[string]$app_name = "",
+	
+	[Parameter(mandatory=$true)]
+	[string]$app_version = "", 
+	
+	[Parameter(mandatory=$true)]
+	[string]$site_hostname = "",
+	
+	[Parameter(mandatory=$true)]
+	[string]$code_folder = ""
+);
+
+## Named constants
+New-Variable -Name "CONTENT_ROOT" -Value "D:\Content"			-Option Constant;
+New-Variable -Name "SITES_ROOT"   -Value "D:\Content\APISites"	-Option Constant;
+New-Variable -Name "BACKUP_ROOT"  -Value "D:\backups"			-Option Constant;
+
+function Main() {
+	## The WebAdministration module requires elevated privileges.
+	$isAdmin = Is-Admin;
+	if( $isAdmin ) {
+		Write-Host -foregroundcolor 'green' "Starting...";
+		Import-Module WebAdministration;
+		CreateSite;
+	} else {
+		Write-Host -foregroundcolor 'red' "This script must be run from an account with elevated privileges.";
+	}
+}
+
+function Is-Admin {
+ $id = New-Object Security.Principal.WindowsPrincipal $([Security.Principal.WindowsIdentity]::GetCurrent());
+ $id.IsInRole([Security.Principal.WindowsBuiltinRole]::Administrator);
+}
+
+function CreateSite() {
+	## Check parameters
+	if ($site_name -eq "")
+	{
+		Write-Host -foregroundcolor 'red' "ERROR: Site Name must not be null.";
+		exit;
+	}
+
+	if ($app_name -eq "")
+	{
+		Write-Host -foregroundcolor 'red' "ERROR: App Name must not be null.";
+		exit;
+	}
+
+	if ($app_version -eq "")
+	{
+		Write-Host -foregroundcolor 'red' "ERROR: App Version must not be null.";
+		exit;
+	}
+
+	if ($site_hostname -eq "")
+	{
+		Write-Host -foregroundcolor 'red' "ERROR: Site Host Name must not be null.";
+		exit;
+	}
+
+	if ($code_folder -eq "")
+	{
+		Write-Host -foregroundcolor 'red' "ERROR: Code Folder must not be null.";
+		exit;
+	}
+
+	## Set up folder names for file system folder structure
+	$sitePath = $SITES_ROOT + "\" + $site_name;
+	$sitePhysicalPath = $sitePath + "\root";
+	$appPath = $sitePath + "\" + $app_name;
+	$versionAppPath = $sitePath + "\" + $app_version;
+
+	## Make site directories
+	## Should check non-existence first
+	foreach ( $folder in (
+			$CONTENT_ROOT,
+			$SITES_ROOT,
+			$sitePath,
+			$sitePhysicalPath,
+			$appPath
+			)
+		)
+	{
+		if (Test-Path $folder)
+		{
+			Write-Host "WARNING: The path ${folder} already exists. Skipping creation.";
+		}
+		else
+		{
+			Write-Host "INFO: Creating folder ${folder}.";
+			New-Item -ItemType directory -Path $folder;     
+		}
+	}
+
+	## Stop web site so updated files can be copied.
+	$webSiteExists = (Get-WebSite -Name $site_name) -ne $null
+	if( $webSiteExists -eq $true ){
+		Stop-Website -Name $site_name
+	}
+	else {
+		write-host "*********** False *************";

```

Comment:
I can't see the overall program output but it seems like this `write-host` line doesn't have enough info or context. Maybe "No website exists to stop" or something like that?